keras_cv/models/backbones/detectron2/detectron2_backbone.py

```diff
@@ -0,0 +1,203 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import copy
+
+from keras_cv.backend import keras
+from keras_cv.layers.detectron2_layers import ViTDetPatchingAndEmbedding
+from keras_cv.layers.detectron2_layers import WindowedTransformerEncoder
+from keras_cv.models.backbones.backbone import Backbone
+from keras_cv.models.backbones.detectron2.detectron2_backbone_presets import (
+    backbone_presets,
+)
+from keras_cv.utils.python_utils import classproperty
+
+
+@keras.utils.register_keras_serializable(package="keras_cv.models")
+class ViTDetBackbone(Backbone):
+    """A ViT image encoder that uses a windowed transformer encoder and
+    relative positional encodings.
+
+    Args:
+        img_size (int, optional): The size of the input image. Defaults to
+            `1024`.
+        patch_size (int, optional): the patch size to be supplied to the
+            Patching layer to turn input images into a flattened sequence of
+            patches. Defaults to `16`.
+        in_chans (int, optional): The number of channels in the input image.
+            Defaults to `3`.
+        embed_dim (int, optional): The latent dimensionality to be projected
+            into in the output of each stacked windowed transformer encoder.
+            Defaults to `1280`.
+        depth (int, optional): The number of transformer encoder layers to
+            stack in the Vision Transformer. Defaults to `32`.
+        mlp_dim (_type_, optional): The dimensionality of the hidden Dense
+            layer in the transformer MLP head. Defaults to `1280*4`.
+        num_heads (int, optional): the number of heads to use in the
+            `MultiHeadAttentionWithRelativePE` layer of each transformer
+            encoder. Defaults to `16`.
+        out_chans (int, optional): The number of channels (features) in the
+            output (image encodings). Defaults to `256`.
+        use_bias (bool, optional): Whether to use bias to project the keys,
+            queries, and values in the attention layer. Defaults to `True`.
+        use_abs_pos (bool, optional): Whether to add absolute positional
+            embeddings to the output patches. Defaults to `True`.
+        use_rel_pos (bool, optional): Whether to use relative positional
+            emcodings in the attention layer. Defaults to `False`.
+        window_size (int, optional): The size of the window for windowed
+            attention in the transformer encoder blocks. Defaults to `0`.
+        global_attention_indices (list, optional): Indexes for blocks using
+            global attention. Defaults to `[7, 15, 23, 31]`.
+        layer_norm_epsilon (int, optional): The epsilon to use in the layer
+            normalization blocks in transformer encoder. Defaults to `1e-6`.
+    """
+
+    def __init__(
+        self,
+        img_size=1024,
```

Comment:
We've standardized on `input_shape` for this in other backbones (accepting a tuple of `(height, width, channels)`)

Reply:
I tried to add a `input_shape` attribute but since I am not using the `Functional` syntax for building the model, it doesn't allow me to add that attribute.

In case you didn't notice, I am using the `call` method to specify the computations in the model instead of passing a symbolic input through each layer/operation in the `__init__` method. 

I noticed that it's just easier not to deal with symbolic inputs with Keras Core. One of the main reasons why Keras Core struggles with symbolic inputs is that it doesn't do shape inference. For example, in Keras, this works:

```python
from tensorflow import keras

x = keras.Input([2, 3])
tf.shape(x)[0] * 10  # Note that even though the shape at axis 0 is None,
                     # TensorFlow returns a symbolic tensor making computations
                     # like these valid instead of throwing an exception.
```

but Keras Core fails, since `x.shape[0]` is `None`.

I think it should not be difficult to convert the implementation to fully use the `Functional` syntax but we would have to manually check if the shapes we are getting are `None`s or not. So, for example, we could do this in Keras Core:

```python
import keras_core
from keras_core import ops

x = keras_core.Input([2, 3])
if x.shape[0] is not None:
    x = ops.reshape(x, (x.shape[0] * 2, 3))
else:
    x = ops.reshape(x, (None, 3))
```

or use some other operation.

Sorry for not highlighting these details properly beforehand! I will add a bunch of comments where I do something unintuitive so it's easier for future reviewers.